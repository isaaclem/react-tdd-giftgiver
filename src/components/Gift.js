import React, { Component } from 'react'
import { 
  Form, 
  FormGroup, 
  FormControl, 
  FormLabel, 
  Button 
} from 'react-bootstrap'

export default class Gift extends Component {
  state = { person: '', present: '' };

  render() {
    return (
      <div>
        <Form>
          <FormGroup>
            <FormLabel>Person</FormLabel>
            <FormControl  
              className='input-person' 
              onChange={event => this.setState({ person: event.target.value })}
            />
          </FormGroup>
          <FormGroup>
            <FormLabel>Present</FormLabel>
            <FormControl  
              className='input-present' 
              onChange={event => this.setState({ present: event.target.value })}
            />
          </FormGroup>
          <Button 
            className='btn-remove'
            onClick={() => this.props.removeGift(this.props.gift.id)}
          >
            Remove Gift
          </Button>
        </Form>
      </div>
    )
  }
}