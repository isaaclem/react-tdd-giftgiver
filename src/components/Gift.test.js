import React from 'react'
import { shallow } from 'enzyme';
import Gift from './Gift.js';
import "../setupTests";

describe('Gifts', () => {
  const mockRemove = jest.fn();
  const id = 1;
  const props = { gift: { id }, removeGift: mockRemove };
  const gift = shallow(<Gift {...props} />);

  it('renders properly', () => {
    expect(gift).toMatchSnapshot(); 
  });

  it('initializes a person and present in `state`', () => {
    expect(gift.state()).toEqual({ person: '', present: ''})
  });

  describe('When typing into the person input', () => {
    const person = 'Uncle';

    beforeEach(() => {
      gift.find('.input-person').simulate('change', { target: { value: person }});
    });

    it('Updates the person in `state`', () => {
      expect(gift.state().person).toEqual(person); 
    });
  });

  describe('When typing into the present input', () => {
    const present = 'Golf club';

    beforeEach(() => {
      gift.find('.input-present').simulate('change', { target: { value: present }});
    });

    it('Updates the present in `state`', () => {
      expect(gift.state().present).toEqual(present);
    });
  });

  describe('when clicking the remove-gift button', () => {
    beforeEach(() => {
      gift.find('.btn-remove').simulate('click');
    });

    it('calls the removeGift callback', () => {
      expect(mockRemove).toHaveBeenCalledWith(id);
    })
  });
});