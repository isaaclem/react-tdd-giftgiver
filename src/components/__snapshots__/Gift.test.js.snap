// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifts renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gift
    gift={
      Object {
        "id": 1,
      }
    }
    removeGift={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ForwardRef
        as="form"
        inline={false}
      >
        <ForwardRef
          as="div"
        >
          <ForwardRef
            column={false}
            srOnly={false}
          >
            Person
          </ForwardRef>
          <ForwardRef
            as="input"
            className="input-person"
            onChange={[Function]}
          />
        </ForwardRef>
        <ForwardRef
          as="div"
        >
          <ForwardRef
            column={false}
            srOnly={false}
          >
            Present
          </ForwardRef>
          <ForwardRef
            as="input"
            className="input-present"
            onChange={[Function]}
          />
        </ForwardRef>
        <ForwardRef
          active={false}
          className="btn-remove"
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          Remove Gift
        </ForwardRef>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "form",
        "children": Array [
          <ForwardRef
            as="div"
          >
            <ForwardRef
              column={false}
              srOnly={false}
            >
              Person
            </ForwardRef>
            <ForwardRef
              as="input"
              className="input-person"
              onChange={[Function]}
            />
          </ForwardRef>,
          <ForwardRef
            as="div"
          >
            <ForwardRef
              column={false}
              srOnly={false}
            >
              Present
            </ForwardRef>
            <ForwardRef
              as="input"
              className="input-present"
              onChange={[Function]}
            />
          </ForwardRef>,
          <ForwardRef
            active={false}
            className="btn-remove"
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            Remove Gift
          </ForwardRef>,
        ],
        "inline": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": Array [
              <ForwardRef
                column={false}
                srOnly={false}
              >
                Person
              </ForwardRef>,
              <ForwardRef
                as="input"
                className="input-person"
                onChange={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Person",
                "column": false,
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Person",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "column": false,
                  "srOnly": false,
                },
                "displayName": "FormLabel",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "input",
                "className": "input-person",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Feedback": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "as": "div",
                    "type": "valid",
                  },
                  "displayName": "Feedback",
                  "propTypes": Object {
                    "as": [Function],
                    "type": [Function],
                  },
                  "render": [Function],
                },
                "defaultProps": Object {
                  "as": "input",
                },
                "displayName": "FormControl",
                "render": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "FormGroup",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": Array [
              <ForwardRef
                column={false}
                srOnly={false}
              >
                Present
              </ForwardRef>,
              <ForwardRef
                as="input"
                className="input-present"
                onChange={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Present",
                "column": false,
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Present",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "column": false,
                  "srOnly": false,
                },
                "displayName": "FormLabel",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "input",
                "className": "input-present",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Feedback": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "as": "div",
                    "type": "valid",
                  },
                  "displayName": "Feedback",
                  "propTypes": Object {
                    "as": [Function],
                    "type": [Function],
                  },
                  "render": [Function],
                },
                "defaultProps": Object {
                  "as": "input",
                },
                "displayName": "FormControl",
                "render": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "FormGroup",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": false,
            "children": "Remove Gift",
            "className": "btn-remove",
            "disabled": false,
            "onClick": [Function],
            "type": "button",
            "variant": "primary",
          },
          "ref": null,
          "rendered": "Remove Gift",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "active": false,
              "disabled": false,
              "type": "button",
              "variant": "primary",
            },
            "displayName": "Button",
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Check": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Input": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "type": "checkbox",
            },
            "displayName": "FormCheckInput",
            "render": [Function],
          },
          "Label": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "type": "checkbox",
            },
            "displayName": "FormCheckLabel",
            "render": [Function],
          },
          "defaultProps": Object {
            "disabled": false,
            "inline": false,
            "isInvalid": false,
            "isValid": false,
            "title": "",
            "type": "checkbox",
          },
          "displayName": "FormCheck",
          "render": [Function],
        },
        "Control": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Feedback": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
              "type": "valid",
            },
            "displayName": "Feedback",
            "propTypes": Object {
              "as": [Function],
              "type": [Function],
            },
            "render": [Function],
          },
          "defaultProps": Object {
            "as": "input",
          },
          "displayName": "FormControl",
          "render": [Function],
        },
        "Group": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "FormGroup",
          "render": [Function],
        },
        "Label": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "column": false,
            "srOnly": false,
          },
          "displayName": "FormLabel",
          "render": [Function],
        },
        "Row": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "FormRow",
          "render": [Function],
        },
        "Text": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "small",
          },
          "displayName": "FormText",
          "render": [Function],
        },
        "defaultProps": Object {
          "as": "form",
          "inline": false,
        },
        "displayName": "Form",
        "render": [Function],
      },
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ForwardRef
          as="form"
          inline={false}
        >
          <ForwardRef
            as="div"
          >
            <ForwardRef
              column={false}
              srOnly={false}
            >
              Person
            </ForwardRef>
            <ForwardRef
              as="input"
              className="input-person"
              onChange={[Function]}
            />
          </ForwardRef>
          <ForwardRef
            as="div"
          >
            <ForwardRef
              column={false}
              srOnly={false}
            >
              Present
            </ForwardRef>
            <ForwardRef
              as="input"
              className="input-present"
              onChange={[Function]}
            />
          </ForwardRef>
          <ForwardRef
            active={false}
            className="btn-remove"
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            Remove Gift
          </ForwardRef>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "form",
          "children": Array [
            <ForwardRef
              as="div"
            >
              <ForwardRef
                column={false}
                srOnly={false}
              >
                Person
              </ForwardRef>
              <ForwardRef
                as="input"
                className="input-person"
                onChange={[Function]}
              />
            </ForwardRef>,
            <ForwardRef
              as="div"
            >
              <ForwardRef
                column={false}
                srOnly={false}
              >
                Present
              </ForwardRef>
              <ForwardRef
                as="input"
                className="input-present"
                onChange={[Function]}
              />
            </ForwardRef>,
            <ForwardRef
              active={false}
              className="btn-remove"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              Remove Gift
            </ForwardRef>,
          ],
          "inline": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": Array [
                <ForwardRef
                  column={false}
                  srOnly={false}
                >
                  Person
                </ForwardRef>,
                <ForwardRef
                  as="input"
                  className="input-person"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Person",
                  "column": false,
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Person",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "column": false,
                    "srOnly": false,
                  },
                  "displayName": "FormLabel",
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "input",
                  "className": "input-person",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "Feedback": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "as": "div",
                      "type": "valid",
                    },
                    "displayName": "Feedback",
                    "propTypes": Object {
                      "as": [Function],
                      "type": [Function],
                    },
                    "render": [Function],
                  },
                  "defaultProps": Object {
                    "as": "input",
                  },
                  "displayName": "FormControl",
                  "render": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
              },
              "displayName": "FormGroup",
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": Array [
                <ForwardRef
                  column={false}
                  srOnly={false}
                >
                  Present
                </ForwardRef>,
                <ForwardRef
                  as="input"
                  className="input-present"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Present",
                  "column": false,
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Present",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "column": false,
                    "srOnly": false,
                  },
                  "displayName": "FormLabel",
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "input",
                  "className": "input-present",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "Feedback": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "as": "div",
                      "type": "valid",
                    },
                    "displayName": "Feedback",
                    "propTypes": Object {
                      "as": [Function],
                      "type": [Function],
                    },
                    "render": [Function],
                  },
                  "defaultProps": Object {
                    "as": "input",
                  },
                  "displayName": "FormControl",
                  "render": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
              },
              "displayName": "FormGroup",
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "active": false,
              "children": "Remove Gift",
              "className": "btn-remove",
              "disabled": false,
              "onClick": [Function],
              "type": "button",
              "variant": "primary",
            },
            "ref": null,
            "rendered": "Remove Gift",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Check": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Input": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "type": "checkbox",
              },
              "displayName": "FormCheckInput",
              "render": [Function],
            },
            "Label": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "type": "checkbox",
              },
              "displayName": "FormCheckLabel",
              "render": [Function],
            },
            "defaultProps": Object {
              "disabled": false,
              "inline": false,
              "isInvalid": false,
              "isValid": false,
              "title": "",
              "type": "checkbox",
            },
            "displayName": "FormCheck",
            "render": [Function],
          },
          "Control": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Feedback": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
                "type": "valid",
              },
              "displayName": "Feedback",
              "propTypes": Object {
                "as": [Function],
                "type": [Function],
              },
              "render": [Function],
            },
            "defaultProps": Object {
              "as": "input",
            },
            "displayName": "FormControl",
            "render": [Function],
          },
          "Group": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "FormGroup",
            "render": [Function],
          },
          "Label": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "column": false,
              "srOnly": false,
            },
            "displayName": "FormLabel",
            "render": [Function],
          },
          "Row": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "FormRow",
            "render": [Function],
          },
          "Text": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "small",
            },
            "displayName": "FormText",
            "render": [Function],
          },
          "defaultProps": Object {
            "as": "form",
            "inline": false,
          },
          "displayName": "Form",
          "render": [Function],
        },
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
